Docs: Add limitations for Basic Authentication

hmanwani-rh opened the pull request:
See JIRA: https://issues.redhat.com/browse/QDOCS-29

Added limitations for Basic Authentication in "security-built-in-authentication.adoc" file. 

Doc preview: http://file.pnq.redhat.com/~hmanwani/QDOCS-29-main

hmanwani-rh committed:
Add limitations for Basic Authentication

sberyozkin commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
+
+There are a few limitations associated with Basic Authentication, which include:
+
+Credentials as plain text::
```
```suggestion
Credentials are sent as plain text::
```

hmanwani-rh replied:
Done, added the suggestion

sberyozkin commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
+--
+It is required to use HTTPS with Basic Authentication. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when HTTP is forwarded to Quarkus.
+
+Also, in multi-hop deployments, HTTPS is only used between client and the first Quarkus endpoint, and the basic credentials are propagated from the first Quarkus endpoint to the next endpoint. Therefore, the credentials are sent as plain text.
```
```suggestion
Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only but they are propagated to the next endpoint over HTTP.
```

hmanwani-rh replied:
Done, changed the sentence as follows:

"Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only. However, the credentials are propagated to the next Quarkus endpoint over HTTP."

sberyozkin commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
+Application complexity increases::
++
+--
+The Quarkus application needs to ensure that username, passwords, and roles are securely secured, which increases the application complexity. 
```
```suggestion
The Quarkus application needs to ensure itself that the usernames, passwords, and roles are managed securely, which increases the application complexity. 
```

hmanwani-rh replied:
Done, added the suggestion.

sberyozkin commented:
Hi @hmanwani-rh Nice work, I've added a few minor suggestions which I thought could clarify a few details better, have a look please

hmanwani-rh commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
+Credentials as plain text::
++
+--
+It is required to use HTTPS with Basic Authentication. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when HTTP is forwarded to Quarkus.
```
Done, added the suggestion.

hmanwani-rh commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
+Also, in multi-hop deployments, HTTPS is only used between client and the first Quarkus endpoint, and the basic credentials are propagated from the first Quarkus endpoint to the next endpoint. Therefore, the credentials are sent as plain text.
+--
+
+Credentials being exposed::
```
Done, added the suggestion

hmanwani-rh commented:
> Hi @hmanwani-rh Nice work, I've added a few minor suggestions which I thought could clarify a few details better, have a look please

@sberyozkin Thanks for reviewing the update and sharing the comments. I've incorporated the changes as suggested.

MichalMaler commented:
> > Hi @hmanwani-rh Nice work, I've added a few minor suggestions which I thought could clarify a few details better, have a look please
> 
> @sberyozkin Thanks for reviewing the update and sharing the comments. I've incorporated the changes as suggested.

Good Job @hmanwani-rh . Will add my review now. Will add just few lines to make it complete.

MichalMaler commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
 The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. Therefore, Basic Authentication is used with HTTPS to provide confidentiality.
- 
+
+There are a few limitations associated with Basic Authentication, which include:
```
```suggestion
Basic Authentication is a well-specified challenge/response scheme that all browsers and most Web servers understand. The main benefit of this method is its ability to work through proxy servers. However, there are a few limitations associated with Basic Authentication, which include:
```

hmanwani-rh replied:
Thanks for sharing this. 
Added this information.

MichalMaler commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
+Application complexity increases::
++
+--
+The Quarkus application needs to ensure itself that usernames, passwords, and roles are managed securely, which increases the application complexity.
```
```suggestion
The Quarkus application needs to ensure that usernames, passwords, and roles are managed securely, which increases the application complexity.
```

hmanwani-rh replied:
I think I'll keep it as is to force on the statement that this actually increases the application complexity.

MichalMaler replied:
OK :)

MichalMaler commented:
LGTM

sberyozkin commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,30 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
- 
+
+
+Basic Authentication is a well-specified challenge or response scheme that all web browsers and most web servers understand. The main benefit of this method is its ability to work through proxy servers. However, there are a few limitations associated with Basic Authentication, which include:
```
```suggestion
Basic Authentication is a well-specified, simple challenge or response scheme that all web browsers and most web servers understand. However, there are a few limitations associated with Basic Authentication, which include:
```

sberyozkin commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,30 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
+--
+It is required to use HTTPS with Basic Authentication to avoid exposing the credentials. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when a request is forwarded to Quarkus over HTTP.
+
+Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only. However, the credentials are propagated to the next Quarkus endpoint over HTTP.
```
```suggestion
Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only and the credentials are propagated to the next Quarkus endpoint over HTTP.
```

hmanwani-rh replied:
Done, added the suggestion.

sberyozkin commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
 The Basic Authentication mechanism does not provide confidentiality protection for the transmitted credentials. The credentials are merely encoded with Base64 when in transit and not encrypted or hashed in any way. Therefore, Basic Authentication is used with HTTPS to provide confidentiality.
- 
+
+Basic Authentication is a well-specified, simple challenge or response scheme that all web browsers and most web servers understand. However, there are a few limitations associated with Basic Authentication, which include:
```
```suggestion
Basic Authentication is a well-specified, simple challenge and response scheme that all web browsers and most web servers understand. However, there are a few limitations associated with Basic Authentication, which include:
```

sberyozkin commented on docs/src/main/asciidoc/security-built-in-authentication.adoc:
```diff
@@ -23,7 +23,29 @@ If the user name is `Alice` and the password is `secret`, the HTTP authorization
+--
+It is required to use HTTPS with Basic Authentication to avoid exposing the credentials. However, if the load balancer terminates HTTPS, the risk of exposing credentials as plain text increases when a request is forwarded to Quarkus over HTTP.
+
+Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only. However, the credentials are propagated to the next Quarkus endpoint over HTTP.
```
```suggestion
Also, in multi-hop deployments, the credentials can be exposed if HTTPS is used between the client and the first Quarkus endpoint only and the credentials are propagated to the next Quarkus endpoint over HTTP.
```

hmanwani-rh replied:
@sberyozkin Thanks for adding that. Fixed the sentence as suggested.

sberyozkin commented:
Thanks @hmanwani-rh I've just squashed a few last commits with `Minor fix` messages...

MichalMaler commented:
> @MichalMaler I kept the last 2 ones as I did not want a reference to your contribution be lost :-), I will squash now to the single commit as looks like you are OK with it :-)

Oh well, hope it will be written forever in our hearts :D :D Thank you for the attitude thought! Well done, We!
